app/scripts/phishing-detect.js

```diff
@@ -3,3 +3,48 @@ window.onload = function() {
       document.getElementById('esdbLink').innerHTML = '<b>To read more about this scam, navigate to: <a href="https://etherscamdb.info/domain/' + window.location.hash.substring(1) + '"> https://etherscamdb.info/domain/' + window.location.hash.substring(1) + '</a></b>'
   }
 }
+
+const dnode = require('dnode')
+const { EventEmitter } = require('events')
+const PortStream = require('extension-port-stream')
+const extension = require('extensionizer')
+const setupMultiplex = require('./lib/stream-utils.js').setupMultiplex
+const { getEnvironmentType } = require('./lib/util')
+const ExtensionPlatform = require('./platforms/extension')
+
+document.addEventListener('DOMContentLoaded', start)
+
+function start () {
+  const windowType = getEnvironmentType(window.location.href)
+
+  global.platform = new ExtensionPlatform()
+  global.METAMASK_UI_TYPE = windowType
+
+  const extensionPort = extension.runtime.connect({ name: windowType })
+  const connectionStream = new PortStream(extensionPort)
+  const mx = setupMultiplex(connectionStream)
+  setupControllerConnection(mx.createStream('controller'), (err, metaMaskController) => {
+    if (err) {
+      return
+    }
+
+    const suspect = window.location.hash.substring(1)
+    const goBack = () => window.history.go(-1)
```

Comment:
I suspect this doesn't work as intended.

Reply:
Hmm, I don't think that'll work if the site isn't using TLS (small number of sites nowadays but still). We might be able to pass the redirect in the hash though, I'll add that.

Reply:
Updated with a better solution, passing the href through to the phishing page